### DISCUSSION TITLE
java.lang.NullPointerException

### DISCUSSION THREAD

**Discussion by kenreiss** [SpenceKonde/megaTinyCore#586] [Q&A]
Sorry to bother, but I've been trying to upgrade to megaTinyCore 2.5.2 and 2.5.4 and always get a red "java.lang.NullPointerException" message at the bottom-left of the Boards Manager window when attempting to either "Upgrade" or select the version then "Install".  I have 2.4.2 installed, which was upgraded previously.  This is running on a rPi Zero W.  

I tried another system with same configuration and I get the same error.  I tried formatting another and fresh install of 2.5.4 works fine, so it's specific to upgrading.  Curious if something in 2.4.2 is causing this.  But it won't even let me downgrade to 2.4.0 without showing the error.  So I tried removing 2.4.2, thinking I would fresh-install 2.5.4, but I get the same Red Java error when I click "Remove" also.  Is there a manual work-around I can try, without having to reformat this entire rPi?  

THANK YOU! Ken.

**Comment by hmeijdam** [comment]
You are not the only one experiencing this issue. [see here](https://github.com/SpenceKonde/megaTinyCore/discussions/567#discussioncomment-1843820)

**Reply by kenreiss** [reply]
Thank you - I'm sorry I hadn't seen your previous post when I searched.  Do you know of any way to merge this post (#586) into yours (#567)?

**Comment by SpenceKonde** [comment]
I think I've identified the screwip that led to this, it was because I seem to not be able to get it htrough my head that no board manaer json file entry must ever be changed once it is released. Even if it seems like you can fix a bug with a tiny change to the json file, no, you won't, all you will do is break the users's ardiino15 folders. And once you make the mistake in one place, there is nothing you can to do undo it, you will only make them problem worse and make it impact more people.

**Reply by kenreiss** [reply]
Thank you!!
